### PR TITLE
Pause docs: `WorldMapArea` enum and related data

### PR DESCRIPTION
**Pull request by Dragorn421**
The `WorldMapArea` enum itself was established by looking at the scene misc. command.
For example Kakariko village: `extracted/gc-eu-mq-dbg/assets/scenes/overworld/spot01/spot01_scene.c`
```c
...
SCENE_CMD_MISC_SETTINGS(0x00, 0x00000001),
...
```
`0x00000001` = 1 is a WorldMapArea value -> `WORLD_MAP_AREA_KAKARIKO_VILLAGE = 1`

### CODE REVIEW

**Review by krm01 — APPROVED** [review]

**Comment by krm01** [comment]
the changes look fine to me, i have no clue how that jenkins failure is related though?

File: include/z64save.h [hunk 1]
```diff
@@ -176,6 +176,32 @@ typedef enum TimerId {
...
+    /* 12 */ WORLD_MAP_AREA_GERUDOS_FORTRESS,
+    /* 13 */ WORLD_MAP_AREA_HAUNTED_WASTELAND,
+    /* 14 */ WORLD_MAP_AREA_MARKET,
+    /* 15 */ WORLD_MAP_AREA_HYRULE_CASTLE,
+    /* 16 */ WORLD_MAP_AREA_DEATH_MOUNTAIN_TRAIL,
+    /* 17 */ WORLD_MAP_AREA_DEATH_MOUNTAIN_CRATER,
+    /* 18 */ WORLD_MAP_AREA_GORON_CITY,
+    /* 19 */ WORLD_MAP_AREA_LON_LON_RANCH,
+    /* 20 */ WORLD_MAP_AREA_20, // hakasitarelay and a lon lon ranch cutscene, gQuestionMarkPositionNameENGTex
+    /* 21 */ WORLD_MAP_AREA_GANONS_CASTLE,
```
Comment (fig02) [inline include/z64save.h:201]: `WORLD_MAP_AREA_20` -> `WORLD_MAP_AREA_QUESTION_MARK` ?
I also had an aneurysm trying to read hakasitarelay cause i didnt realize it was referencing an og scene name.
Maybe `// Used by Windmill/Dampe Race and a Lon Lon Ranch Cutscene` ?

what cutscene btw? why does a cutscene have a unique map area, you cant pause?

Comment (Dragorn421) [reply include/z64save.h:201]: I don't remember which cutscene, and yeah it's probably a mistake as you can't pause in that scene layer
I can change to questionmark yeah

Comment (mzxrules) [reply include/z64save.h:201]: fun fact: this enum is annoyingly close to matching the SPOT_XX order. The only difference is there is no SPOT_14 (14 maps to Market), and Lon Lon is SPOT_20 not 19.

**Comment by fig02** [comment]
fails on oot-gc-jp, bss?

**Review by cadmic — APPROVED** [review]

**Review by fig02 — APPROVED** [review]

**Pull request merged**